crmprtd/ec_swob/download.py

```diff
@@ -43,6 +43,7 @@ def get_url_list(
 
     # Match something like this: 2019-10-18-1600-bc-env-asw-1a02p-AUTO-swob.xml
     search_pattern = re.compile(r"{}.*swob\.xml".format(date.strftime("%Y-%m-%d-%H")))
+    search_pattern_dfo = re.compile(rf"{date.strftime('%Y%m%dT%H30Z')}.*SWOB.*\.xml")
```

Comment:
OK, so *here* is the spot where you actually *do* want to ignore case, if you were to incorporate these two patterns into one and match either "SWOB" or "swob". It might be a bit tricky, but I'd propose something like this:

```python3
datep_standard = datetime.strftime("%Y-%m-%d-%H")
datep_dfo = datetime.strftime("%Y%m%dT%H30Z")
search_pattern = re.compile(rf"({datep_standard}|{datep_dfo}).*swob.*\.xml", re.IGNORECASE)
```

The `(pattern_a|pattern_b)` *should* match either pattern. See the `|` special character in the [Regular Expression Syntax docs](https://docs.python.org/3/library/re.html#regular-expression-syntax).

Reply:
Given the trickiness here, it would probably be worth breaking this URL matching out into its own function so that we can test the pattern matching independently.

Reply:
I've moved the pattern matching out and added a few minimal test cases.